Base/Python/slicer/util.py

```diff
@@ -2395,6 +2395,139 @@ def createProgressDialog(parent=None, value=0, maximum=100, labelText="", window
       setattr(progressIndicator, key, value)
   return progressIndicator
 
+
+from contextlib import contextmanager
+
+@contextmanager
+def displayPythonShell(display=True):
+  """Show the Python console while the code in the context manager is being run.
+
+  The console stays visible only if it was visible already.
+
+  :param show: If show is False, the context manager has no effect.
+
+  .. code-block:: python
+
+    with slicer.util.displayPythonShell():
+      slicer.util.pip_install('nibabel')
+
+  """
+  import slicer
+
+  def dockableWindowEnabled():
+    import qt
+    return toBool(qt.QSettings().value("Python/DockableWindow"))
+
+  def getConsole():
+    return mainWindow().pythonConsole().parent() if dockableWindowEnabled() else pythonShell()
+
+  if display:
+    console = getConsole()
+    consoleVisible = console.visible
+    console.show()
+    slicer.app.processEvents()
+  try:
+    yield
+  finally:
+    if display:
+      console.setVisible(consoleVisible)
+
+class WaitCursor:
+  def __init__(self, show=True):
+    """Set the cursor to waiting mode while the code in the context manager is being run.
+
+    :param show: If show is False, this context manager has no effect.
+
+    .. code-block:: python
+
+      import time
+
+      with slicer.util.WaitCursor():
+        time.sleep(2)
+    """
+    self.show = show
+
+  def __enter__(self):
+    import qt, slicer
+    if self.show:
+      qt.QApplication.setOverrideCursor(qt.Qt.WaitCursor)
+      slicer.app.processEvents()
+
+  def __exit__(self, type, value, traceback):
+    if self.show:
+      import qt
+      qt.QApplication.restoreOverrideCursor()
+
+class MessageDialog:
+  def __init__(self, message, show=True, logLevel='INFO'):
```

Comment:
```suggestion
  def __init__(self, message, show=True, logLevel=logging.INFO):
```

Reply:
I thought this would be best as well, but didn't go for it because it needs to `import logging` outside the definition and for some reason (why??) that doesn't happen in `slicer.util`.